src/adaptive_threshold.jl

```diff
@@ -40,58 +35,66 @@ the width and height. You can use the convenience function
 `recommend_size(::AbstractArray{T,2})` to obtain this suggested value.
 If left unspecified, a default value of 32 is utilised.
 
-#Example
+# References
+1. Bradley, D. (2007). Adaptive Thresholding using Integral Image. *Journal of Graphic Tools*, 12(2), pp.13-21. [doi:10.1080/2151237x.2007.10129236](https://doi.org/10.1080/2151237x.2007.10129236)
 
-```julia
-using TestImages
+See also: [`binarize`](@ref ImageBinarization.binarize), [`BinarizationAlgorithm`](@ref ImageBinarization.BinarizationAlgorithm)
+"""
+struct AdaptiveThreshold <: BinarizationAlgorithm
+  window_size::Int
+  percentage::Int
+end
+AdaptiveThreshold(; percentage::Int = 15, window_size::Int = 32) = AdaptiveThreshold(percentage, window_size)
 
-img = testimage("cameraman")
-s = recommend_size(img)
-binarize(AdaptiveThreshold(percentage = 15, window_size = s), img)
```

Comment:
I removed all these example codes, but now I guess there's a need to hold them even though it is quite trivial.
wrt https://github.com/zygmuntszpak/ImageBinarization.jl/issues/26, new example might goes like this:
```julia
using TestImages
img = testimage("cameraman")
s = recommend_size(img)
filter = AdaptiveThreshold(percentage = 15, window_size = s)
filter(img)
```